### PR TITLE
Log Fragment name when trying to render a lazy fragment

### LINKED ISSUE #26910: Bug: React fails to log invariant 306 message when lazy() resolves to a `Fragment`
React version: 18.2.0

## Steps To Reproduce

1. Try to render this component: `const LazyFragment = lazy(() => Promise.resolve({ default: Fragment }));`

Link to code example:

https://github.com/tom-sherman/react-lazy-fragment-bug

## The current behavior

It throws an error:

```
Cannot convert a Symbol value to a string
```

## The expected behavior

It should throw an error related to invariant 306 eg.

```
Element type is invalid. Received a promise that resolves to: Fragment. Lazy element type must resolve to a class or function.
```

### PR DESCRIPTION
**Pull request by tom-sherman**
## Summary

Fixes #26910 

## How did you test this change?

See test.

### CODE REVIEW

File: packages/react-reconciler/src/__tests__/ReactLazy-test.internal.js [hunk 1]
```diff
@@ -757,6 +757,32 @@ describe('ReactLazy', () => {
...
+    await waitForAll(['Loading...']);
+
+    await resolveFakeImport(React.Fragment);
+    root.update(
+      <Suspense fallback={<Text text="Loading..." />}>
+        <BadLazy />
+      </Suspense>,
+    );
+    await waitForThrow(
+      'Element type is invalid. Received a promise that resolves to: <Fragment>. ' +
```
Comment (eps1lon) [inline packages/react-reconciler/src/__tests__/ReactLazy-test.internal.js:781]: It doesn't resolve to `<Fragment>` though but `Fragment`.

File: packages/react-reconciler/src/ReactFiberBeginWork.js [hunk 2]
```diff
@@ -1891,11 +1892,13 @@ function mountLazyComponent(
     }
   }
 
+  const loggedComponent = Component === REACT_FRAGMENT_TYPE ? '<Fragment>' : Component;
```
Comment (eps1lon) [inline packages/react-reconciler/src/ReactFiberBeginWork.js:1895]: We can use `getComponentNameFromType` from `shared/getComponentNameFromType` here instead. That way we also get a better name for all the other types you're not supposed to use.
```suggestion
  const loggedComponent = getComponentNameFromType(Component) || String(Component);
```

**Review by eps1lon — APPROVED** [review]
Thanks

File: packages/react-reconciler/src/ReactFiberBeginWork.js (hunk 2 above)
Comment (eps1lon) [reply packages/react-reconciler/src/ReactFiberBeginWork.js]: Forgot that we use `+ ''` for perf so `|| (Component + '')` it is.

**Comment by eps1lon** [comment]
Prettier and lint failing.

**Comment by tom-sherman** [comment]
@eps1lon I chose to ignore the string coercion lint, let me know if that's not ok and I can find a different way forward - probably by introducing a new string coercion check function for components?

**Comment by eps1lon** [comment]
Let's just be simple about it and do what the lint rule asks.

**Comment by eps1lon** [comment]
Seems like common practice though.

**Comment by tom-sherman** [comment]
> Let's just be simple about it and do what the lint rule asks.

What solution are you thinking of? The lint rule asks to use a "checkXxxxxStringCoercion" function but none exists for components right now, which is why I suggested creating one.

**Review by eps1lon — APPROVED** [review]

**Comment by eps1lon** [comment]
> What solution are you thinking of? 

Just kept the original behavior.

**Pull request merged**